### ISSUE TITLE
problem installing on CentOS

### ISSUE THREAD

**Issue by matteofcorona** [IBM-Cloud/ibm-cloud-developer-tools#35]
Hi,
just trying to install on CentOS:

# curl -sL https://ibm.biz/idt-installer | bash
[main] --==[ IBM Cloud Developer Tools for Linux/MacOS - Installer, v1.2.3 ]==--
[main] WARN: Linux has only been tested on Ubuntu, please let us know if you use this utility on other Distros
[install] Starting Full Installation...
[install_deps] Checking for and updating 'apt-get' support on Linux
which: no apt-get in (/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/root/bin)
[install_deps] ERROR: 'apt-get' is not found.  Thats the only linux installer I know, sorry.

Is there a way to install on Centos ?
Thank you

**Comment by kfbishop** [comment]
Hi @matteofcorona - Currently, we've only coded with `apt-get` based distros.  It will be a while before I can load up, code and test for CentOS, but I'll put it on my Todo list.   I wouldn't expect many roadblocks to make this work.  Google shows there are ways to install apt on centos, but this could lead to an ugly untested rabbit hole, but if you want to explore....    
PRs / hints, guidance are always welcomed too!

You should be able to manually install all the pieces yourself.
1. install dependencies: curl, git, docker, kubectl, helm
1. Install IBM Cloud CLI:   https://console.bluemix.net/docs/cli/index.html
1. Install required "bx" plugins:  cloud-functions, container-registry, container-service, dev, schematics, sdk-gen
    - `bx plugin install <plugin>`
1. Optional: create launcher `idt` shell, which simply calls `bx dev $*`.   See towards the end of the installer where this is defined (Line ~377)

If you do these steps above, please take notes, as it will help me extend the real installer to support CentOS.  Also, if you run into other issues, let me know and I'll try to assist.  cheers.

**Comment by gibfahn** [comment]
> Google shows there are ways to install apt on centos, but this could lead to an ugly untested rabbit hole, but if you want to explore....

I think `yum` would be an easier way to go, it should just be a case of `if ubuntu|debian then apt install foo; elif centos|rhel then yum install foo`.

@a-roberts would probably be a good guinea-pig for this as well (as a RHEL user).

**Comment by hhue13** [comment]
Waiting for RHEL / CentOS version as well ...

**Comment by vikramkhatri** [comment]
Your support for only Ubuntu is making life difficult for IBM Cloud Private - ICP Knowledge Center link https://www.ibm.com/support/knowledgecenter/SSBS6K_2.1.0.3/manage_cluster/install_cli.html redirects for bx install to https://console.bluemix.net/docs/cli/index.html#overview and which turns out to be only applicable to Ubuntu.

Majority of Enterprise S/W runs on RHEL and not Ubuntu - it is difficult to know how would one install bx for RHEL/CentOS as links are buried in your Ubuntu only installer.

Until you find time to support RHEl/CentOS, please update the https://console.bluemix.net/docs/cli/index.html to give the direct download link instead of writing `error "'apt-get' is not found.  Thats the only linux installer I know, sorry."`

**Comment by triceam** [comment]
You are correct, the installation script currently only supports Ubuntu.  We've had several requests recently to support RHEL & CentOS, so we're considering those options.   We're also actively working to update content in the docs to make things clearer. 

In the meantime, you can manually install the core `ibmcloud` CLI using the steps listed here:  
https://console.bluemix.net/docs/cli/reference/bluemix_cli/get_started.html#getting-started

Then you can install the `dev` plugin and prerequisites following these instructions:  https://console.bluemix.net/docs/cli/ts_createapps.html#appendix

**Comment by wanghaifeng** [comment]
Here is my steps on install bluemix api on a blue cloud server

`yum install  kubernetes-client etcd git curl
 
# Install helm binary file, the folder linux-amd64 can be removed manually
curl https://storage.googleapis.com/kubernetes-helm/helm-v2.11.0-linux-amd64.tar.gz | tar -zxvf - && mv linux-amd64/helm /usr/local/bin
 
 
# Install bluemix-cli, the URL cannot be connected on blue cloud server
curl https://plugins.stage1.ng.bluemix.net/download/bluemix-cli/latest/linux64 -o /tmp/bluemix-cli.tgz
tar zxvf /tmp/bluemix-cli.tgz
chmod 755 /tmp/Bluemix_CLI/install
/tmp/Bluemix_CLI/install`
